runbotics/runbotics-scheduler/src/nest-mailer/nest-mailer.module.ts

```diff
@@ -0,0 +1,32 @@
+import { Module } from '@nestjs/common';
+import { MailerModule } from '@nestjs-modules/mailer';
+import { NestMailerService } from './nest-mailer.service';
+import { UserModule } from '#/database/user/user.module';
+import { Logger } from '#/utils/logger';
+
+@Module({
+    imports: [
+        MailerModule.forRootAsync({
+            useFactory: () => ({
+                transport: {
+                    host: process.env.MAIL_HOST,
+                    port: Number(process.env.MAIL_PORT),
+                    secure: false, // upgrade later with STARTTLS
+                    tls: { rejectUnauthorized: false },
+                    auth: {
+                        user: process.env.MAIL_USERNAME,
+                        pass: process.env.MAIL_PASSWORD,
+                    },
+                },
+                defaults: {
+                    from: process.env.MAIL_USERNAME,
+                },
+            }),
+        }),
```

Comment:
use ServerConfigService

Reply:
does it work even if we don't provide those envs? or what is happening in this scenario?

Reply:
Can I use ServerConfigService here while dynamically initializing MailerModule with forRootAsync method? One idea I have right now is to make NestMailerModule as dynamic module and then inject ServerConfigService there and use its methods. What do you think about this approach?

Reply:
About env variables, I tested it and all errors are handled by `.catch()` while executing ` await this.mailerService.sendMail()`. All errors, including those caused by invalid env variables will be caught there safely and then displayed as error message.

Reply:
here is what you need:
![image](https://github.com/runbotics/runbotics/assets/44121335/6afa36ec-8d90-415e-a1c0-5d98aa444c4b)


> Can I use ServerConfigService here while dynamically initializing MailerModule with forRootAsync method? One idea I have right now is to make NestMailerModule as dynamic module and then inject ServerConfigService there and use its methods. What do you think about this approach?